BEIMA.Backend.Test/DeviceTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿using BEIMA.Backend.MongoService;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using Moq;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace BEIMA.Backend.Test
+{
+    [TestFixture]
+    public class DeviceTest : UnitTestBase
+    {
+
+        #region FailureTests
+
+        [Test]
+        public async Task IdNotInDatabase_GetDevice_ReturnsNotFound()
+        {
+            // ARRANGE
+
+            // Setup mock database client.
+            Mock<IMongoConnector> mockDb = new Mock<IMongoConnector>();
+            var testGuid = "1234567890abcdef1234567890abcdef";
+            mockDb.Setup(mock => mock.GetDevice(It.Is<ObjectId>(oid => oid == new ObjectId(testGuid))))
+                  .Returns<BsonDocument>(null)
+                  .Verifiable();
```

Comment:
This can be used as a sort of reference for how to do a mock client. You can set individual methods to return specific values depending on the input parameters.